src/models/CreateAndRunTest.ts

```diff
@@ -0,0 +1,270 @@
+
+    CoreUtils.exportVariable(PostTaskParameters.runId, testRunResult.testRunId);
+
+    await awaitTerminations(testRunResult.testRunId, apiService, ValidationType.testrun);
+    testRunResult = await awaitResultsPopulation(testRunResult.testRunId, apiService) ?? testRunResult;
+    CoreUtils.exportVariable(PostTaskParameters.isRunCompleted, 'true');
+
+    printMetrics(testRunResult);
+    await uploadResultsToPipeline(testRunResult);
+    setTaskResults(testRunResult);
+    setOutputVariable(testRunResult);
+}
+
+async function uploadAllTestFiles(yamlModel: YamlConfig, apiService: APIService) : Promise<void> {
+    let configFiles = yamlModel.configurationFiles;
+    if(configFiles.length > 0) {
+        console.log("Uploading the configuration files");
+        await uploadFiles(configFiles, apiService, FileType.ADDITIONAL_ARTIFACTS);
+        console.log(`Uploaded ${configFiles.length} configuration file(s) for the test successfully.`);
+    }
+
+    let zipFiles = yamlModel.zipArtifacts;
+    if(zipFiles.length > 0) {
+        console.log("Uploading the zip artifacts");
+        await uploadFiles(zipFiles, apiService, FileType.ZIPPED_ARTIFACTS);
+        console.log(`Uploaded ${zipFiles.length} zip artifact(s) for the test successfully.`);
+    }
+
+    let propertyFile = yamlModel.propertyFile;
+    if(propertyFile != undefined && propertyFile!= '') {
+        console.log("Uploading the user properties file");
+        await apiService.uploadFile(propertyFile, FileType.USER_PROPERTIES);
+        console.log(`Uploaded user properties file for the test successfully.`);
+    }
+
+    await uploadTestScriptFile(yamlModel, apiService);
+
+    return;
+}
+
+async function uploadFiles(configFiles: string[], apiService: APIService, fileType: FileType) : Promise<void> {
+    if(configFiles.length > 0) {
+        for(let filepath of configFiles){
+            await apiService.uploadFile(filepath, fileType);
+        };
+    }
+    return;
+}
+
+async function uploadTestScriptFile(yamlModel: YamlConfig, apiService: APIService) : Promise<void> {
+    let testScriptFile = yamlModel.testPlan;
+    let fileType = FileType.TEST_SCRIPT;
+    if(yamlModel.kind == TestKind.URL){
+        fileType = FileType.URL_TEST_CONFIG;
+    }
+    console.log("Uploading the test script file");
+    await apiService.uploadFile(testScriptFile, fileType);
+    console.log(`Uploaded test script file for the test successfully.`);
+    return;
+}
+
+function printPortalUrl(testRunDisplayName: string, testDisplayName: string, subscriptionName: string, resourceId: string) : void {
+    console.log("\nView the load test run in Azure portal by following the steps:");
+    console.log("1. Go to your Azure Load Testing resource '"+Util.getResourceNameFromResourceId(resourceId)+"' in subscription '"+subscriptionName+"'");
+    console.log("2. On the Tests page, go to test '"+testDisplayName+"'");
+    console.log("3. Go to test run '"+testRunDisplayName+"'\n");
```

Comment:
Just curious, we do send the portal url in the test run object, why dont we use it here?

Reply:
github sanitises that and the link wont work.